.github/workflows/code-quality.yml

```diff
@@ -26,8 +39,16 @@ jobs:
         with:
           persist-credentials: false
 
+      - name: Update Adapters and Core branches
+        if: ${{ github.event_name == 'workflow_call' || github.event_name == 'workflow_dispatch'}}
+        shell: bash
+        run: |
+          ./.github/scripts/update_dev_packages.sh \
+            ${{ github.event.inputs.dbt_adapters_branch }} \
+            "main"
+
       - name: Setup `hatch`
-        uses: dbt-labs/dbt-adapters/.github/actions/setup-hatch@main
+        uses: dbt-labs/dbt-adapters/.github/actions/setup-hatch@${{ inputs.dbt_adapters_branch }}
```

Comment:
My guess is that you can't do this on a PR since `inputs` won't exist. That's why the code quality check isn't running. That being said, I think you would always want the `setup-hatch` workflow on `main` anyway, and this can be removed.

Reply:
good catch, updated to use GHA's funky ternary syntax to default to main in those cases

Reply:
so turns out it's a little weirder, you can't actually use an expression in `uses`, see: https://github.com/orgs/community/discussions/27048